### PR TITLE
Initial draft for define data elements

### PR DESCRIPTION
**Pull request by phuo1032**

### CODE REVIEW

File: places-launch-rules/define-data-elements.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Defining Data Elements
+
+### Data Elements
+
+Data elements are the building blocks for the applications data dictionary and are used to collect, organize, and deliver data across marketing and ad technology.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: - Please add an apostrophe in "applications" --> "application's".

File: places-launch-rules/define-data-elements.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+A data element is a variable where the value can be mapped to a Visitor ID, a Carrier Name, an Advertising ID, a Push ID and so on. In Experience Platform Launch, you can reference this value by its variable name. This collection of data elements becomes the dictionary of defined data that you can use to build your rules \(events, conditions, and actions\), and this dictionary is shared across Experience Platform Launch where it can be used with any extension in your property.
+
+With Places extension, you can reference values from the following targets:
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please add "the" between "With" and "Places"

File: places-launch-rules/define-data-elements.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* Current POI, which refers to the POI in which your customer is currently located. If the user is located in multiple POIs, the one belongs to the library with higher rank is picked. If multiple POIs belong to the highest rank library, the one with the smallest radius is picked.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please move the content starting with the 2nd sentence down one line and line it up with the bullet. It should look like this:
"* Current POI, which refers to the POI in which your customer is currently located. 

    If the user is located in multiple POIs, the one belongs to the library with higher rank is picked. If multiple POIs belong to the highest rank library, the one with the smallest radius is picked.

File: places-launch-rules/define-data-elements.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+* Last exited POI, which refers to the most recent POI that the user exited.
+* Last entered POI, which refers to the most recent POI that the user entered. 
+
+Each POI contains the following data references
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please add a colon (:) after "references"

File: places-launch-rules/define-data-elements.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+* Longitude: longitude of the POI
+* Metadata: custom metadata of the POI
+* Name: region of the POI
+* Radius: radius of the POI
+* Region ID: ID of the POI
+* Region/State: region, province or state of the POI
+
+### Create a data element
+
+1. From a Property page, open the Data Elements tab.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please change "From a Property page" to:
"In the Property page for your app, click the **Data Elements** tab.

File: places-launch-rules/define-data-elements.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+![Create DE 1](../.gitbook/assets/create-de-1.png)
+
+2. Click **Create New Data Element**.
+![Create DE 2](../.gitbook/assets/create-de-2.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+3. From the extension menu, select Places.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please reword to:
"3. In the list of installed extensions, find Places."

File: places-launch-rules/define-data-elements.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+![Create DE 3](../.gitbook/assets/create-de-3.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+4. From the Data Element Type drop-down list, select a data reference for this data element.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please reword to 
"4. In the **Data Element Type** drop-down list, select a data reference for this data element."

File: places-launch-rules/define-data-elements.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+![Create DE 4](../.gitbook/assets/create-de-4.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+5. Select a POI target.
+![Create DE 5](../.gitbook/assets/create-de-5.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+6. If this data element is a custom metadata reference, pick a metadata key.
+![Create DE 6](../.gitbook/assets/create-de-6.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+7. Type a name for the data element, and click Save .
+![Create DE 7](../.gitbook/assets/create-de-7.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove image link, not necessary

File: places-launch-rules/define-data-elements.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+3. From the extension menu, select Places.
+![Create DE 3](../.gitbook/assets/create-de-3.png)
+
+4. From the Data Element Type drop-down list, select a data reference for this data element.
+![Create DE 4](../.gitbook/assets/create-de-4.png)
+
+5. Select a POI target.
+![Create DE 5](../.gitbook/assets/create-de-5.png)
+
+6. If this data element is a custom metadata reference, pick a metadata key.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please replace "pick" with "select"

File: places-launch-rules/define-data-elements.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+![Create DE 6](../.gitbook/assets/create-de-6.png)
+
+7. Type a name for the data element, and click Save .
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please remove the comma, put "Save" in bold, and remove the extra space between "Save" and the period. Should look like this:
"7. Type a name for the data element and click **Save**.

File: places-launch-rules/define-data-elements.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+![Create DE 7](../.gitbook/assets/create-de-7.png)
+
+
+### Use a data element
+
+After a data element is created, you can use it from any rule component if a data element picker is present.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please reword as follows:
"After a data element is created, if a data element picker is present, you can use the data element from any rule component."

File: places-launch-rules/define-data-elements.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+![Use DE 1](../.gitbook/assets/use-de-1.png)
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: please remove the 2 image links...not necessary

File: places-launch-rules/define-data-elements.md [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+![Use DE 2](../.gitbook/assets/use-de-2.png)
+
+If a data element picker is not present in the rule component, you may still use the data element by wrapping the data element name with the **%%** tokens.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: Please replace "may still" with "can"

File: places-launch-rules/define-data-elements.md [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+For example, if the data elmement name is **Last POI City**, you may add **%%LAST POI City%%** to a text input.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md:55]: Please remove "may" and replace with "can"

File: places-launch-rules/define-data-elements.md [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+
+
+### Publish data elements
+
+If data elements are used in any of the rule components. They must also be included in the library and published.
```
Comment (rekrishn) [inline places-launch-rules/define-data-elements.md]: The first sentence is not a complete sentence. It looks like the 2 sentences should be combined like this:
"If data elements are used in any of the rule components, these data elements must also be included in the library and published."

**Review by rekrishn — CHANGES_REQUESTED** [review]
some changes @phuo1032...let me know if you have any questions.

**Comment by phuo1032** [comment]
@rekrishn I updated the text as requested. I think the screenshots are helpful for customers to onboard quickly, even with the cost of adding difficulties to localize later. Also, Launch is not localized so the screenshot will pretty much be the same.

Let's get some input from others to see what they think.

**Comment by rekrishn** [comment]
> @rekrishn I updated the text as requested. I think the screenshots are helpful for customers to onboard quickly, even with the cost of adding difficulties to localize later. Also, Launch is not localized so the screenshot will pretty much be the same.
> 
> Let's get some input from others to see what they think.

**Review by rekrishn — APPROVED** [review]
looks good @phuo1032, thanks!

**Pull request merged**